presctl/cmd/entities_test.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"testing"
+
+	"github.com/Microsoft/presidio/presctl/cmd/entities"
+)
+
+// clientMock is a mock for http client
+type clientMock struct {
+	returningCode int
+}
+
+// TestIsJson tests that the method differentiate between a valid to a non valid json string
+func TestIsJson(t *testing.T) {
+	nonJSONStr := "A simple string"
+	if isJSON(nonJSONStr) {
+		t.Fatal("isJson failed. the given string is not a valid JSON, yet the method returned true")
+	}
+
+	nonJSONStr = `{
+	                itemA: "A"
+                  }`
+	if isJSON(nonJSONStr) {
+		t.Fatal("isJson failed. the given string is not a valid JSON, yet the method returned true")
+	}
+
+	validJSONStr := `{
+	            	  "itemA": "A"
+                     }`
+	if !isJSON(validJSONStr) {
+		t.Fatal("isJson failed. the given is a valid JSON, yet the method returned false")
+	}
+}
+
+// TestFailingCreateTemplate tests that when presidio returns an invalid status code the cli exists with an error
+func TestFailingCreateTemplate(t *testing.T) {
+	// we expect the exit code to be 1, if the returned status code is not as expected (not 201)
+	if os.Getenv("BE_CRASHER") == "1" {
```

Comment:
It looks like this test is dependant on external resource (env. variable).
In an isolated test this is not guaranteed to work every time.
A way to solve it is to mock `os.Getenv` to return this requested value.

Reply:
this is a trick to test the EXIT code of the cli process.
notice what happens. it start with value 0, and re enter with value 1. so it can actually start an external process (the cli) and then check the code. the mock wont help here....

Reply:
Can you add a clear explanation about it somewhere?
At least for me it was not clear, perhaps for others as well.